R/table.R

```diff
@@ -172,7 +172,10 @@ kable_caption = function(label, caption, format) {
   # create a label for bookdown if applicable
   if (is.null(label)) label = opts_current$get('label')
   if (is.null(label)) label = NA
-  if (!is.null(caption) && !is.na(caption) && !is.na(label)) caption = paste0(
+  is_na = function(v) {
+    length(v) == 0 || (length(v) == 1 && is.na(v))
+  }
+  if (!is.null(caption) && !is_na(caption) && !is_na(label)) caption = paste0(
```

Comment:
No longer need to test for `is.null()` since `is_na(NULL)` is `TRUE`.

```suggestion
  if (!is_na(caption) && !is_na(label)) caption = paste0(
```